RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -206,63 +206,332 @@ namespace lst {
+    if (slope12 > 0 and slope23 < 0) {
+      if (x1 < x2 and x2 < x3)
+        charge = 1;
+      else if (x1 > x2 and x2 > x3)
+        charge = 1;
+      else if (y1 < y2 and y2 < y3)
+        charge = -1;
+      else if (y1 > y2 and y2 > y3)
+        charge = -1;
+    } else if (slope12 < 0 and slope23 > 0) {
+      if (x1 < x2 and x2 < x3)
+        charge = -1;
+      else if (x1 > x2 and x2 > x3)
+        charge = -1;
+      else if (y1 < y2 and y2 < y3)
+        charge = 1;
+      else if (y1 > y2 and y2 > y3)
+        charge = 1;
+    } else if (slope12 > 0 and slope23 > 0) {
+      if (slope12 > slope23)
+        charge = 1;
+      else
+        charge = -1;
+    } else if (slope12 < 0 and slope23 < 0) {
+      if (slope12 > slope23)
+        charge = 1;
+      else
+        charge = -1;
+    }
+
+    //get the absolute value of px and py at the initial point
+    float pseudo_phi = alpaka::math::atan(
+        acc, (y_init - y_center) / (x_init - x_center));  //actually represent pi/2-phi, wrt helix axis z
+    float Px = Pt * alpaka::math::abs(acc, alpaka::math::sin(acc, pseudo_phi)),
+          Py = Pt * alpaka::math::abs(acc, cos(pseudo_phi));
+
+    //Above line only gives you the correct value of Px and Py, but signs of Px and Py calculated below.
+    //We look at if the circle is clockwise or anti-clock wise, to make it simpler, we separate the x-y plane into 4 quarters.
+    if (x_init > x_center && y_init > y_center)  //1st quad
+    {
+      if (charge == 1)
+        Py = -Py;
+      if (charge == -1)
+        Px = -Px;
+    }
+    if (x_init < x_center && y_init > y_center)  //2nd quad
+    {
+      if (charge == -1) {
+        Px = -Px;
+        Py = -Py;
+      }
+    }
+    if (x_init < x_center && y_init < y_center)  //3rd quad
+    {
+      if (charge == 1)
+        Px = -Px;
+      if (charge == -1)
+        Py = -Py;
+    }
+    if (x_init > x_center && y_init < y_center)  //4th quad
+    {
+      if (charge == 1) {
+        Px = -Px;
+        Py = -Py;
+      }
+    }
+
+    //But if the initial T5 curve goes across quarters(i.e. cross axis to separate the quarters), need special redeclaration of Px,Py signs on these to avoid errors
+    if (x3 < x2 && x2 < x1)
+      Px = -alpaka::math::abs(acc, Px);
+    if (x3 > x2 && x2 > x1)
+      Px = alpaka::math::abs(acc, Px);
+    if (y3 < y2 && y2 < y1)
+      Py = -alpaka::math::abs(acc, Py);
+    if (y3 > y2 && y2 > y1)
+      Py = alpaka::math::abs(acc, Py);
+
+    float AO = alpaka::math::sqrt(acc, (x_other - x_center) * (x_other - x_center) + (y_other - y_center) * (y_other - y_center)); 
+    float BO = alpaka::math::sqrt(acc, (x_init - x_center) * (x_init - x_center) + (y_init - y_center) * (y_init - y_center));
+    float AB2 = (x_other - x_init) * (x_other - x_init) + (y_other - y_init) * (y_other - y_init); 
+    float dPhi = alpaka::math::acos(acc, (AO * AO + BO * BO - AB2) / (2 * AO * BO)); //Law of Cosines
+    float ds = circleRadius / 100 * dPhi;
```

Comment:
Yeah we just wanted to be consistent with the T5 helix calculation. @YonsiG any reason we need to use meters instead of centimeters for the calculations?

Reply:
I think our calculation are based on standard units like, Tesla, meters, speed of light... so that was a safety check for calculation. Actually it doesn't matter if you convert everything also into centimeter, but I made it consistent in unit so the calculation can be more easily guaranteed correct.